wasm-host/src/host_functions.rs

```diff
@@ -480,3 +482,128 @@ pub fn oracle_keylet(
     set_data(keylet_hash.len() as i32, out_buf_ptr, keylet_hash, _caller)?;
     Ok(vec![WasmValue::from_i32(HASH256_LEN as i32)])
 }
+
+pub fn trace(
+    _data_provider: &mut DataProvider,
+    _inst: &mut Instance,
+    _caller: &mut CallingFrame,
+    inputs: Vec<WasmValue>,
+) -> Result<Vec<WasmValue>, CoreError> {
+    // Expect 5 inputs.
+
+    // check the number of inputs
+    if inputs.len() != 5 {
+        return Err(CoreError::Execution(CoreExecutionError::FuncSigMismatch));
+    }
+
+    let msg_read_ptr = if inputs[0].ty() == ValType::I32 {
```

Comment:
Test what this function does without this error check when there aren't enough function inputs supplied by the WASM call (e.g., trace).

See what happens when the type is incorrect (e.g., supply a buffer but expect a number)